agent/lib_mongodb.c

```diff
@@ -173,7 +175,166 @@ NR_PHP_WRAPPER(nr_mongodb_operation) {
 }
 NR_PHP_WRAPPER_END
 
-void nr_mongodb_enable(TSRMLS_D) {
+#else
+
+NR_PHP_WRAPPER(nr_mongodb_operation_before) {
+  (void)wraprec;
+  nr_segment_t* segment;
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+  segment->wraprec = auto_segment->wraprec;
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_mongodb_operation_after) {
+  const char* this_klass = "MongoDB\\Operation\\Executable";
+  zval* collection = NULL;
+  zval* database = NULL;
+  zval* server = NULL;
+  zval* this_var = NULL;
+  bool discard_segment = false;
+  nr_datastore_instance_t instance = {
+      .host = NULL,
+      .port_path_or_id = NULL,
+      .database_name = NULL,
+  };
+
+  // tell the compiler to ignore the cast from const char * to char *
+  // to save having to do a strdup operation
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+  nr_segment_datastore_params_t params = {
```

Comment:
Probably the other [nr_segment_datastore_params_t](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_segment_datastore.h#L17) values (like collection and database_name) should be explicitly initialized to NULL here.

Reply:
`database_name` is already set to `NULL` in the `instance` variable declaration above.
`collection` set to `NULL` here f2759015630151982029c4db88d6d71c3229e311